scalameta/trees/shared/src/main/scala/scala/meta/Trees.scala

```diff
@@ -87,10 +87,26 @@ object Lit {
   // 1.4f.toString == "1.399999976158142" // in JS
   // 1.4f.toString == "1.4"               // in JVM
   // See https://www.scala-js.org/doc/semantics.html#tostring-of-float-double-and-unit
-  @ast class Double(format: scala.Predef.String) extends Lit { val value = format.toDouble }
-  object Double { def apply(double: scala.Double): Double = Lit.Double(double.toString) }
-  @ast class Float(format: scala.Predef.String) extends Lit { val value = format.toFloat }
-  object Float { def apply(float: scala.Float): Float = Lit.Float(float.toString) }
+  @ast class Double(
+      format: scala.Predef.String,
+      @newField("4.9.0") number: scala.Double = scala.Double.NaN
```

Comment:
1. we can't use just the string representation because we'd have to duplicate the implementation of a parser dealing with separators
2. we can't use just the double field because the string representation might differ wildly between platforms, jdk/js versions etc., and someone has already changed to string before
3. what we could try using is BigDecimal but that also contains several fields inside, and will require changes in how quasiquotes "lift" these values (they are currently not supported).